vast/vast.cpp

```diff
@@ -67,10 +67,15 @@ try_handle_deprecations(vast::system::default_configuration& cfg) {
   if (caf::holds_alternative<bool>(cfg, "vast.use-legacy-query-scheduler"))
     VAST_WARN("the 'vast.use-legacy-query-scheduler' option no longer exists "
               "and will be ignored.");
-  if (caf::get_or(cfg, "vast.store-backend", "segment-store") == "archive") {
+  if (caf::get_or(cfg, "vast.store-backend", "feather") == "archive") {
     VAST_WARN("the 'vast.store-backend' option 'archive' is deprecated; "
-              "automatically using 'segment-store' instead");
-    caf::put(cfg.content, "vast.store-backend", "segment-store");
+              "automatically using 'feather' instead");
+    caf::put(cfg.content, "vast.store-backend", "feather");
+  } else if (caf::get_or(cfg, "vast.store-backend", "feather")
+             == "segment-store") {
+    VAST_WARN("the 'vast.store-backend' option 'segment-store' is deprecated; "
+              "automatically using 'feather' instead");
+    caf::put(cfg.content, "vast.store-backend", "feather");
```

Comment:
This may or may not need to be split off in a separate PR, depending on the results of ongoing discussion.

Reply:
Yeah, we'll likely vote on it tomorrow. In case of an undecided result I'll create a follow-up story for discussion.

Reply:
We discussed this between @tobim @mavam and myself today, and came to the agreement to drop the segment-store as-is in this PR alongside bumping the partition version with the next minor release. We consider the advantage of being able to drop the segment store with VAST v3.0 as greater than the potential for error. That resulted in a vote of 3-1 for making the removal.